File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/mobs/Bandit.java

package com.watabou.pixeldungeon.actors.mobs;

import com.nyrds.pixeldungeon.ai.MobAi;
import com.nyrds.pixeldungeon.ai.ThiefFleeing;
import com.watabou.pixeldungeon.Dungeon;
import com.watabou.pixeldungeon.actors.Char;
import com.watabou.pixeldungeon.actors.CharUtils;
import com.watabou.pixeldungeon.actors.buffs.Blindness;
import com.watabou.pixeldungeon.actors.buffs.Buff;
import com.watabou.utils.Random;

import org.jetbrains.annotations.NotNull;


public class Bandit extends Thief {

	@Override
	public int attackProc(@NotNull Char enemy, int damage) {
		if (CharUtils.steal(this, enemy)) {
			setState(MobAi.getStateByClass(ThiefFleeing.class));
			Buff.prolong(getEnemy(), Blindness.class, Random.Int(5, 12));
			enemy.observe();
		}
		return damage;
	}
}

File: scripts/mobs/Bandit.lua
local RPD = require "scripts/lib/commonClasses"
local mob = require "scripts/lib/mob"

-- java Bandit (batch 7): a thief that also blinds the victim it robbed
-- from (its defenceProc gold drop is inherited from Thief).
return mob.init{
    attackProc = function(self, enemy, dmg)
        if enemy ~= nil and RPD.CharUtils:steal(self, enemy) then
            RPD.setAi(self, "ThiefFleeing")
            -- Random.Int(5,12) is [5,12)
            RPD.Buffs.Buff:prolong(enemy, "Blindness", math.random(5, 11))
            enemy:observe()
        end
        return dmg
    end,

    defenceProc = function(self, enemy, dmg)
        if self:getState():getTag() == "THIEFFLEEING" then
            RPD.item("Gold", 1):doDrop(self)
        end
        return dmg
    end,
}
